pkg/binlog-filter/filter.go

```diff
@@ -96,6 +96,10 @@ type BinlogEvent struct {
 
 // NewBinlogEvent returns a binlog event filter
 func NewBinlogEvent(rules []*BinlogEventRule) (*BinlogEvent, error) {
+	if len(rules) == 0 {
+		return nil, nil
+	}
+
```

Comment:
so now, we can not create a Filter with no rules? (we can't split New and AddRule ?)

Reply:
review it again after I finish it